quipucords/scanner/network/inspect_callback.py

```diff
@@ -1,239 +1,298 @@
-            "msg", "No information given on unreachable warning."
-        )
-        message = f"UNREACHABLE {host}. {result_message}"
-        self.scan_task.log_message(message, log_level=logging.ERROR)
-        self._finalize_host(host, SystemInspectionResult.UNREACHABLE)
-
-    def event_callback(self, event_dict=None):  # noqa: C901
-        """Control the event callback for Ansible Runner."""
+        https://ansible.readthedocs.io/projects/runner/en/stable/python_interface/?highlight=callback#runner-event-handler
+        TODO: find documentation for these events.
+        """
+        logger.debug("processing event")
+        logger.debug("event_dict=%s", event_dict)
+        okay = ["runner_on_ok", "runner_item_on_ok"]
+        failed = ["runner_on_failed", "runner_item_on_failed"]
+        unreachable = ["runner_on_unreachable"]
+        runner_ignore = [
+            "runner_on_skipped",
+            "runner_item_on_skipped",
+            "runner_on_start",
+        ]
         try:
-            logger.debug("processing event callback")
-            logger.debug("event_dict=%s", event_dict)
-            okay = ["runner_on_ok", "runner_item_on_ok"]
-            failed = ["runner_on_failed", "runner_item_on_failed"]
-            unreachable = ["runner_on_unreachable"]
-            runner_ignore = ["runner_on_skipped", "runner_item_on_skipped"]
-            ignore_states = ["runner_on_start"]
-            event = event_dict.get("event")
-            event_data = event_dict.get("event_data")
-
-            if not event:
-                self.scan_task.log_message(
-                    log_messages.TASK_UNEXPECTED_FAILURE
-                    % ("event_callback", event, event_dict),
-                    log_level=logging.ERROR,
-                )
-                return
-
-            if event in ignore_states:
-                return
-
-            if event_dict:
-                # Check if it is a task event
-                if event.startswith("runner"):
-                    if event in okay:
-                        self.task_on_ok(event_dict)
-                    elif event in failed:
-                        self.task_on_failed(event_dict)
-                    elif event in unreachable:
-                        self.task_on_unreachable(event_dict)
-                    elif event not in runner_ignore:
-                        self.scan_task.log_message(
-                            log_messages.TASK_UNEXPECTED_FAILURE
-                            % ("event_callback", event, event_dict),
-                            log_level=logging.ERROR,
-                        )
-                # Save last role for task logging later
-                if event == "playbook_on_task_start" and event_data:
-                    event_role = event_data.get("role")
-                    if event_role != self.last_role:
-                        self.last_role = event_role
-        except Exception as err_msg:  # noqa: BLE001
```

Comment:
removed this generic try/except because the one wrapping `ansible_runner.run` in `InspectTaskRunner` already covers something unexpected here (and I would rather avoid having these blind excepts everywhere)